students/kimc05/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+import sys
+#Paths and File Processing
+import pathlib
+
+pth = pathlib.Path("./")
+pth.is_dir()
+
+#get path
+pth.absolute()
+
+#Christine Kim
+#Python210 Lesson 3 Mailroom Part 1
+
+#Donor dictionary created
+givetree = {("Rutherford", "Cullen"): (1500, 4200, 50000),
+            ("Theirin", "Alistair"): (200, 80000, 1500000),
+            ("Arainai", "Zevran"): (50,),
+            ("Amell", "Solona"): (2, 500000, 2000000),
+            ("Lavellan", "Soufehla"): (70, 600)}
+
+
+#Prompt for user to be displayed
+prompt = ("\nWelcome to the Blight Orphans Charity. Plase select from the following options.\n"
+                "1: Send a Thank You\n"
+                "2: Create a Report\n"
+                "3: Send a Letter to All\n"
+                "4: Quit\n")
+
+#method for menu selection
+def menu(prompt, dispatch_dict):
+    while True:
+        #receive user input
+        response = input(prompt)
+        if response not in ["1", "2", "3", "4"]:
+            print("Please choose from 1, 2, 3 or 4")
+        #direct user to proper function
+        elif dispatch_dict[response]() == "exit menu":
+            #Quit Script
+            break
```

Comment:
see guidelines for mailroom: https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#controlling-main-program-flow

ideally you want the quit function to call sys.exit() and this way you can utilize the switch dict where you don't have special logic.